### PR TITLE
Date tests

### PR DESCRIPTION
**Pull request by moa114**
Unit tests for date_search_string.py
Also includes two small changes in the date_search_string.py that were discovered during testing:
- using date instead of datetime when checking holiday range
- functionality for if the user does not have a country

### CODE REVIEW

**Review by zrohacova — APPROVED** [review]

File: homeassistant/components/spotify/date_search_string.py [hunk 1]
```diff
@@ -210,8 +210,9 @@ def get_day_of_week(self, current_date: date):
     def search_string_date(self, hass: HomeAssistant, user: Any):
         """Generate a search string for the date feature, if there is no holiday, the current season, month and day is returned, otherwise the current holiday."""
         current_date = dt_util.now().date()
-        country = user["country"]
-
+        country = " "
+        if user is not None and "country" in user:
```
Comment (amalev18) [inline homeassistant/components/spotify/date_search_string.py]: What should happen if the country key isn't found in user? Should this be taken care of in the method calling this method solely?

File: homeassistant/components/spotify/date_search_string.py [hunk 2]
```diff
@@ -210,8 +210,9 @@ def get_day_of_week(self, current_date: date):
     def search_string_date(self, hass: HomeAssistant, user: Any):
         """Generate a search string for the date feature, if there is no holiday, the current season, month and day is returned, otherwise the current holiday."""
         current_date = dt_util.now().date()
-        country = user["country"]
-
+        country = " "
```
Comment (amalev18) [inline homeassistant/components/spotify/date_search_string.py]: Why is this necessary?

File: homeassistant/components/spotify/date_search_string.py (hunk 1 above)
Comment (moa114) [reply homeassistant/components/spotify/date_search_string.py]: Good point. I guess it is good to check it here as well to allow for further extension. I fix!

File: homeassistant/components/spotify/date_search_string.py (hunk 2 above)
Comment (moa114) [reply homeassistant/components/spotify/date_search_string.py]: to have something to send into get_season even if the user did not have country information, but this is outdated now when I have added checks that the user has country

**Review by amalev18 — APPROVED** [review]

**Review by zrohacova — APPROVED** [review]
